xbmc/interfaces/json-rpc/PlaylistOperations.cpp

```diff
@@ -212,6 +212,87 @@ JSONRPC_STATUS CPlaylistOperations::Swap(const std::string &method, ITransportLa
   return ACK;
 }
 
+JSONRPC_STATUS CPlaylistOperations::SetShuffle(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  int playlist = GetPlaylist(parameterObject["playlistid"]);
+  CVariant shuffle = parameterObject["shuffle"];
+  CGUIWindowSlideShow *slideshow = NULL;
+  switch (playlist)
+  {
+    case PLAYLIST_MUSIC:
+    case PLAYLIST_VIDEO:
+    {
+      if (CServiceBroker::GetPlaylistPlayer().IsShuffled(playlist))
+      {
+        if ((shuffle.isBoolean() && !shuffle.asBoolean()) ||
+            (shuffle.isString() && shuffle.asString() == "toggle"))
```

Comment:
You could extract the second part of this condition into a variable outside of the `switch-case` block to avoid having to write it four times.

Reply:
I created two variables outside of the `switch-case`:

```cpp
bool shuffle = (parameterObject["shuffle"].isBoolean() && parameterObject["shuffle"].asBoolean()) ||
               (parameterObject["shuffle"].isString() && parameterObject["shuffle"].asString() == "toggle");
bool unshuffle = (parameterObject["shuffle"].isBoolean() && !parameterObject["shuffle"].asBoolean()) ||
                 (parameterObject["shuffle"].isString() && parameterObject["shuffle"].asString() == "toggle");
```

Reply:
That's better but `(parameterObject["shuffle"].isString() && parameterObject["shuffle"].asString() == "toggle")` still appears twice in both variable definitions.

Reply:
A new proposal with three variables:

```cpp
bool shuffle = parameterObject["shuffle"].isBoolean() && parameterObject["shuffle"].asBoolean();
bool unshuffle = parameterObject["shuffle"].isBoolean() && !parameterObject["shuffle"].asBoolean();
bool toggle = parameterObject["shuffle"].isString() && parameterObject["shuffle"].asString() == "toggle";
```

Reply:
That's better, thanks!